src/main/java/dev/dokan/dokan_java/Unsigned.java

```diff
@@ -10,6 +10,182 @@
 import static java.lang.annotation.ElementType.TYPE_USE;
 
 
+/**
+ * This annotation is used used to indicate that a value with an integer type or an integer type pointer refers to an <b>unsigned value.</b><br>
+ * In this case integers are <i>numbers without positions after decimal point.</i>
+ * (Java's Default integer types being {@code byte, short, int, long} and their corresponding wrappers.)
+ * <h1>Introduction</h1>
+ * Java stores integer types in <a href="https://en.wikipedia.org/wiki/Two%27s_complement">Two's complement-Representation.</a>
+ * Usually numbers represented as Two's complement use the Most-Significant-Bit (MSB) to store the sign of the number.
+ * If a field is annotated with <i>@Unsigned</i> this rule <b>does not apply!</b>
+ * Instead the MSB should be considered as part of the number itself, resulting in a doubled storage capacity of the field
+ * while <b>removing</b> the support for signed values (the field becomes unsigned and the value should only be interpreted as positive.)
+ * See this table as reference for the resulting differences when interpreting numbers:
+ * <table>
+ * <thead>
+ *   <tr>
+ *     <th>Signed value (Java Default)</th>
+ *     <th>Unsigned values (@Unsigned)</th>
+ *     <th>Bitmask (for a byte)</th>
+ *   </tr>
+ * </thead>
+ * <tbody>
+ *   <tr>
+ *     <td>0</td>
+ *     <td>0</td>
+ *     <td>0000 0000</td>
+ *   </tr>
+ *   <tr>
+ *     <td>1</td>
+ *     <td>1</td>
+ *     <td>0000 0001</td>
+ *   </tr>
+ *   <tr>
+ *     <td>2</td>
+ *     <td>2</td>
+ *     <td>0000 0010</td>
+ *   </tr>
+ *   <tr>
+ *     <td>64</td>
+ *     <td>64</td>
+ *     <td>0100 0000</td>
+ *   </tr>
+ *   <tr>
+ *     <td>-128</td>
+ *     <td>128</td>
+ *     <td>1000 0000</td>
+ *   </tr>
+ *   <tr>
+ *     <td>-127</td>
+ *     <td>129</td>
+ *     <td>1000 0001</td>
+ *   </tr>
+ *   <tr>
+ *     <td>-1</td>
+ *     <td>255</td>
+ *     <td>1111 1111</td>
+ *   </tr>
+ * </tbody>
+ * </table>
+ * <h1>Usage</h1>
+ * As Java only supports signed numbers, the correct interpretation and handling of unsigned numbers must be dealt with by the developer.
+ * Because of this it's <b>strongly recommended</b> to tag all values that are unsigned as such <b>by using this annotation.</b><br>
+ * <i>@Unsigned</i> is defined to {@link Target target} {@link java.lang.annotation.ElementType#TYPE_PARAMETER} and
+ * {@link java.lang.annotation.ElementType#TYPE_USE} and can therefore be applied to all usages of types (type contexts)
+ * and parameterized types respectively. See <a href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-9.html#jls-9.6.4.1">&sect9.6.4.1</a>
+ * and <a href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-4.html#jls-4.11">&sect4.11</a> of "The Java&reg Language Specification" for
+ * further reference.<br>
+ * <br>
+ * At least the following cases ("minimum usage") should be annotated with <i>@Unsigned</i> to
```

Comment:
In the future it may be useful to check our/foreign source code for compliance to this rule, e.g. with [the Checker Framework.](https://github.com/typetools/checker-framework)

Reply:
// Do not close this conversation //